[Flaking Test] k8s.io/kubernetes/test/integration/scheduler: nominated_node_name

### Which jobs are flaking?

https://testgrid.k8s.io/sig-release-master-blocking#integration-master

### Which tests are flaking?

`k8s.io/kubernetes/test/integration/scheduler: nominated_node_name`

### Since when has it been flaking?

[05/05/2026, 11:46](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/136833/pull-kubernetes-integration/2051599368208781312)
[04/05/2026, 2:25](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2051095600283258880)

### Testgrid link

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2051095600283258880

### Reason for failure (if possible)

```
{Failed  === RUN   TestSchedulerRestartWithNominatedNode
    nominated_node_name_test.go:861: Unexpected NominatedNodeName after scheduler abort for pod pod-a, got node-other, want node-preferred
    nominated_node_name_test.go:893: pod-a scheduled on node-other, wanted node-preferred
    nominated_node_name_test.go:894: pod-b scheduled on node-preferred, wanted node-other
    test_server.go:270: Skipping metrics scrape because context is already canceled: context canceled
--- FAIL: TestSchedulerRestartWithNominatedNode (3.28s)
}
```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig scheduling

--- a/test/integration/scheduler/nominated_node_name/nominated_node_name_test.go
+++ b/test/integration/scheduler/nominated_node_name/nominated_node_name_test.go
@@ -828,17 +828,23 @@ func TestSchedulerRestartWithNominatedNode(t *testing.T) {
 		pls = append(pls, mockQueueSort, mockPreBind)
 		pls = append(pls, additionalPlugins...)
 		registry, prof := schedulerutils.InitRegistryAndConfig(t, nil, pls...)
-		return schedulerutils.InitTestSchedulerForFrameworkTest(t, testContext, 0,
+		testCtx, teardown := schedulerutils.InitTestSchedulerForFrameworkTest(t, testContext, 0,
 			false, /* do not run scheduler immediately after init */
 			scheduler.WithProfiles(prof),
 			scheduler.WithFrameworkOutOfTreeRegistry(registry),
 		)
+		// Ensure both nodes are in the cache before proceeding
+		if err := testutils.WaitForNodesInCache(testCtx.Ctx, testCtx.Scheduler, 2); err != nil {
+			t.Fatalf("Failed to wait for nodes in cache: %v", err)
+		}
+		return testCtx, teardown
 	}
 
 	// To allow triggering scheduler restart in binding phase, we need to initialize the scheduler
 	// and pass the testCtx.SchedulerCloseFn to the bind plugin before calling Scheduler.Run.
 	mockBind := &mockBindPlugin{}
 	testCtx, _ := initSched(mockBind)
+
 	// This will fail the binding cycle and stop the scheduler
 	// Scheduler should set NNN for podA before this step.
 	mockBind.bindFn = func() *fwk.Status {

